framework/py/flwr/common/object_store/object_store_test.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2025 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tests for ObjectStore."""
+
+
+import unittest
+from abc import abstractmethod
+
+from .object_store import ObjectStore
+
+
+class ObjectStoreTest(unittest.TestCase):
+    """Test all ObjectStore implementations."""
+
+    # This is to True in each child class
+    __test__ = False
+
+    @abstractmethod
+    def object_store_factory(self) -> ObjectStore:
+        """Provide ObjectStore implementation to test."""
+        raise NotImplementedError()
+
+    def test_get_non_existent_key(self) -> None:
+        """Test get method with a non-existent key."""
+        object_store = self.object_store_factory()
+        key = "non_existent_key"
+
+        retrieved_value = object_store.get(key)
+
+        self.assertIsNone(retrieved_value)
+
+    def test_put_and_get(self) -> None:
+        """Test put and get methods."""
+        object_store = self.object_store_factory()
+        key = "test_key"
+        value = b"test_value"
+
+        object_store.put(key, value)
+        retrieved_value = object_store.get(key)
+
+        self.assertEqual(value, retrieved_value)
+
+    def test_put_overwrite(self) -> None:
+        """Test put method with an existing key."""
+        object_store = self.object_store_factory()
+        key = "test_key"
+        value1 = b"test_value1"
+        value2 = b"test_value2"
+
+        object_store.put(key, value1)
+        object_store.put(key, value2)
+        retrieved_value = object_store.get(key)
+
+        self.assertEqual(value2, retrieved_value)
+
+    def test_put_empty_key(self) -> None:
```

Comment:
I wonder if we should prevent this from succeeding and only allow if a non-empty `str` key is passed.